WordPress/Classes/Models/AbstractPost.m

```diff
@@ -452,8 +452,17 @@ - (BOOL)isDraft
 
 - (void)publishImmediately
 {
-    self.dateModified = [NSDate date];
-    [self setDateCreated:self.dateModified];
+    // If the post was modified later than it was created
+    // then this is a post that was published at some point
+    // we will not update the dateCreated
+    if (self.dateCreated &&
```

Comment:
It looks like this change can conflict with the intended functionality of the Publish Immediately option in the Post Options > Date picker depending on the current value of `self.dateModified`.
<img width="371" alt="screen shot 2017-05-03 at 11 31 34 am" src="https://cloud.githubusercontent.com/assets/1435271/25670998/1ca23a5e-2ff4-11e7-8141-b56029f8d984.png">
When testing, I started with a post published on March 17 and set its status to Draft and saved.  I then edited the post, went to the Options screen and tapped Publish Immediately.  Only the dateModified was changed.